### PR TITLE
tests: add calldata to calldatacopy tests

### PR DESCRIPTION
**Pull request by roynalnaruto**
This PR adds tx input to the witness data for `CALLDATACOPY` tests.

This is added so as to not miss any constraint/failure that could've been possible due to calldata being empty. At the moment, the copy circuit implementation is OK.

### CODE REVIEW

File: zkevm-circuits/src/copy_circuit.rs
Comment (z2trillion) [inline zkevm-circuits/src/copy_circuit.rs:796]: You can get the same effect by doing `assert_eq(res, Ok(()))` instead of `assert_eq(res.is_ok())`

File: zkevm-circuits/src/copy_circuit.rs
Comment (z2trillion) [inline zkevm-circuits/src/copy_circuit.rs:820]: I think it would be useful to add `dbg!(length)` after this, so that in case the test fails, we can reproduce it.

**Review by z2trillion — CHANGES_REQUESTED** [review]

File: zkevm-circuits/src/copy_circuit.rs
Comment (CPerezz) [reply zkevm-circuits/src/copy_circuit.rs:820]: I'd use a known and easy to identify number such as `0xcafe`.
See: https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/641

On that way, will be less required to debug stuff because the random values are known beforehand and the tests are easier to reproduce

**Review by CPerezz — COMMENTED** [review]
LGTM, but first wold like that the comments are addressed before merging.

**Review by ed255 — APPROVED** [review]
LGTM modulo the unresolved requests.

**Review by CPerezz — APPROVED** [review]
LGTM modulo the tests that are failing fixed and passing

**Review by z2trillion — APPROVED** [review]

**Pull request merged**
